3scale/3scale-operator
# THREESCALE-8749: Ability to set APICAST_SERVICE_CACHE_SIZE

## Description

Add optional fields in the Apicast production and staging sections of the APIManager CRD that sets the `APICAST_SERVICE_CACHE_SIZE` environment variable in the Apicast containers

## Verification steps

On a cluster, run the operator and create an APIManager with the new fields set. Check that once reconciled the `DeploymentConfigs` include the environment variable with the value from the new fields:

1. Apply the manifests
    ```sh
    make install
    ``` 
2. Run the operator
    ```sh
    make run
    ```
3. Create an APIManager that includes the following in the spec:
    ```yaml
    apicast:
      productionSpec:
        serviceCacheSize: 20
      stagingSpec:
        serviceCacheSize: 10
    ```
4. Check the DeploymentConfigs for APICast
    ```sh
    oc get dc/apicast-staging -o yaml
    oc get dc/apicast-production -o yaml
    ```
5. Verify that they include the environment variable
    ```yaml
    # apicast-staging
    - name: APICAST_SERVICE_CACHE_SIZE
       value: '10'

    # apicast-production
    - name: APICAST_SERVICE_CACHE_SIZE
       value: '20'
    ```
